src/main/java/seedu/module/model/ModuleBook.java

```diff
@@ -87,7 +95,13 @@ public void addTask(Task p) {
      * The task identity of {@code editedTask} must not be the same as another existing task in the module book.
      */
     public void setTask(Task target, Task editedTask) {
-        requireNonNull(editedTask);
+        if (ModuleManager.getListOfExistingModules().contains(editedTask.getModule().toString())) {
```

Comment:
think there is too much method chaining going on so may be good to change this implementation? Up to u if you think otherwise.